Source/player.cpp

```diff
@@ -1973,17 +1974,9 @@ void CheckCheatStats(Player &player)
 void Player::CalcScrolls()
 {
 	_pScrlSpells = 0;
-	for (int i = 0; i < _pNumInv; i++) {
-		if (!InvList[i].isEmpty() && (InvList[i]._iMiscId == IMISC_SCROLL || InvList[i]._iMiscId == IMISC_SCROLLT)) {
-			if (InvList[i]._iStatFlag)
-				_pScrlSpells |= GetSpellBitmask(InvList[i]._iSpell);
-		}
-	}
-
-	for (auto &item : SpdList) {
-		if (!item.isEmpty() && (item._iMiscId == IMISC_SCROLL || item._iMiscId == IMISC_SCROLLT)) {
-			if (item._iStatFlag)
-				_pScrlSpells |= GetSpellBitmask(item._iSpell);
+	for (Item &item : InventoryAndBeltPlayerItemsRange { *this }) {
+		if (item.IsScroll() && item._iStatFlag) {
+			_pScrlSpells |= GetSpellBitmask(item._iSpell);

```

Comment:
Can't we use something from algorithm to do this without an explicit foreach too (like we did for counting)?

Reply:
It is possible with `std::accumulate` but is less readable because we also filter and there isn't an `accumulate_if`

Reply:
This could use a custom accumulator function, something like (untested):

```cpp
const InventoryAndBeltPlayerItemsRange items { *this };
_pScrlSpells = std::accumulate(items.begin(), items.end(), 0, [](uint64_t mask, const Item &item) {
    if (item.IsScroll() && item._iStatFlag) {
        mask |= GetSpellBitmask(item._iSpell);
    }
    return mask;
});
```

Don't think it's worth it for this case unless you really want to use `<algorithm>` concepts.